lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -202,12 +212,45 @@ def _on_start(self, event: StartEvent):
             event.defer()
             return
 
+        self.status.clear(TLSNotFullyConfigured)
+
```

Comment:
I don't see why it shouldn't be possible, but I also don't see a valid reason for it I think. If I understand correctly, you're essentially postponing the whole validation / configuration of the cluster to the `start_opensearch` instead of NOT emitting a start unless it qualified as `startable`, correct? in which case we still would defer in non-leader units in case `deployment_desc is None`. Did I miss something? 

*For the last bullet point, non-leaders do not start from `*-changed event`, they start from a deferred `_on_start`